src/wp-includes/user.php

```diff
@@ -682,14 +682,25 @@ function count_many_users_posts( $users, $post_type = 'post', $public_only = fal
 		return $pre;
 	}
 
-	$userlist = implode( ',', array_map( 'absint', $users ) );
-	$where    = get_posts_by_author_sql( $post_type, true, null, $public_only );
+	// Cleanup the users array. Remove duplicates and sort for consistent ordering.
+	$users = array_unique( array_filter( array_map( 'absint', (array) $users ) ) );
+	sort( $users );
 
-	$result = $wpdb->get_results( "SELECT post_author, COUNT(*) FROM $wpdb->posts $where AND post_author IN ($userlist) GROUP BY post_author", ARRAY_N );
+	$userlist    = implode( ',', $users );
+	$cache_key   = 'count_many_users_posts:' . md5( implode( ':', array( $post_type, $public_only ? 'public' : 'public|private', $userlist, get_current_user_id() ) ) );
+	$cache_salts = array( wp_cache_get_last_changed( 'posts' ), wp_cache_get_last_changed( 'users' ) );
+	$count       = wp_cache_get_salted( $cache_key, 'user-queries', $cache_salts );
```

Comment:
```suggestion
	$count       = wp_cache_get_salted( $cache_key, 'post-queries', $cache_salts );
```
This is a query on the post table, so we should use that group.

Reply:
Thanks, I doubled check and yes, `post-queries` is the right way